### PR TITLE
feat: create delete account page

### PR DESCRIPTION
**Pull request by LinaYahya**
image

image

### CODE REVIEW

File: src/components/main/DeleteMemberDialog.tsx [hunk 1]
```diff
@@ -19,48 +20,83 @@ type Props = {
...
-        <DialogContent>
-          <DialogContentText id={alertDialogDescription}>
+        <IconButton
+          aria-label="close"
+          onClick={closeModal}
+          sx={{
+            position: 'absolute',
+            right: 8,
+            top: 8,
+            color: (theme) => theme.palette.grey[500],
```
Comment (spaenleh) [inline src/components/main/DeleteMemberDialog.tsx:57]: Default color here is fine. I don't think there is a big difference. If it is too light people might think that it is disabled

File: src/components/main/DeleteMemberDialog.tsx [hunk 2]
```diff
@@ -19,48 +20,83 @@ type Props = {
...
   return (
     <>
       <Dialog
         open={open}
-        onClose={() => setOpen(false)}
+        onClose={closeModal}
         aria-labelledby={alertDialogTitle}
         aria-describedby={alertDialogDescription}
+        maxWidth="sm"
+        fullWidth
```
Comment (spaenleh) [inline src/components/main/DeleteMemberDialog.tsx:45]: Add `disableRestoreFocus` here and `autoFocus` on the TextField

File: src/components/main/DeleteMemberDialog.tsx [hunk 3]
```diff
@@ -19,48 +20,83 @@ type Props = {
...
+
         <DialogActions>
-          <Button onClick={() => setOpen(false)} color="primary">
+          <Button onClick={closeModal} color="primary">
             {translateAccount('PROFILE_DELETE_ACCOUNT_MODAL_CANCEL_BUTTON')}
           </Button>
           <Button
             onClick={() => deleteMember({ id })}
             color="error"
             autoFocus
```
Comment (spaenleh) [inline src/components/main/DeleteMemberDialog.tsx:89]: Remove this.

File: src/components/main/DeleteMemberDialog.tsx [hunk 4]
```diff
@@ -19,48 +20,83 @@ type Props = {
...
+            {translateAccount('PROFILE_DELETE_TYPE_CONFIRMATION_TEXT', {
+              text: confirmationDeleteTextToCompare,
+            })}
+          </DialogContentText>
+          <TextField
+            value={confirmationDeleteValue}
+            fullWidth
+            variant="outlined"
+            placeholder={confirmationDeleteTextToCompare}
+            onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (spaenleh) [inline src/components/main/DeleteMemberDialog.tsx:76]: The extra typing here does not look like it is necessary.

File: src/components/main/DeleteMemberDialog.tsx [hunk 5]
```diff
@@ -19,48 +20,83 @@ type Props = {
...
-      <Grid
-        container
-        spacing={3}
-        direction="column"
-        alignItems="flex-start"
-        my={1}
-      >
+      <Grid container direction="column" alignItems="flex-start" my={1}>
         <Grid item xs={8}>
           <Grid item xs={12}>
```
Comment (spaenleh) [inline src/components/main/DeleteMemberDialog.tsx:101]: Not sure about the utility of the grids here...

File: src/config/paths.ts [hunk 6]
```diff
@@ -6,3 +6,4 @@ export const PAYMENT_CONFIRM_PATH = '/payment/confirm';
 export const PASSWORD_SETTINGS_PATH = '/password';
 export const AVATAR_SETTINGS_PATH = '/avatar';
 export const PUBLIC_PROFILE_PATH = '/public-profile';
+export const DELETE_ACCOUNT_PATH = '/delete-account';
```
Comment (spaenleh) [inline src/config/paths.ts:9]: I think this route should be changes, if possible to something without a hyphen. 
As we might want to make the page more general, maybe `/manage` ?

**Review by spaenleh — CHANGES_REQUESTED** [review]
Thank you for this PR ! 
🔥 

Design wise, I would like that new page to be more generic. Think of it as the "Danger zone" where definitive actions can be done to the account and such. So I think:
- update list item on the sidebar should read "Destructive settings" or "Danger Zone"
- the page should present a more welcoming display (not just a simple button)
- Add a red frame around the dangerous actions to warn people they are entering a special place
- use the same style for the title as on the other pages (this should be uniform)
- rename screens and paths to follow the general naming you choose in 1.

File: src/components/main/DeleteMemberDialog.tsx [hunk 7]
```diff
@@ -19,48 +20,83 @@ type Props = {
...
         open={open}
-        onClose={() => setOpen(false)}
+        onClose={closeModal}
         aria-labelledby={alertDialogTitle}
         aria-describedby={alertDialogDescription}
+        maxWidth="sm"
+        fullWidth
       >
-        <DialogTitle id={alertDialogTitle}>
+        <DialogTitle id={alertDialogTitle} sx={{ paddingBottom: 1 }}>
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/components/main/DeleteMemberDialog.tsx:47]: Both are done to reduce padding between title and dialog content

File: src/components/main/DeleteMemberDialog.tsx [hunk 8]
```diff
@@ -14,83 +15,83 @@ import { mutations } from '../../config/queryClient';
...
-const DeleteMemberDialog = ({ id }: Props): JSX.Element => {
+const DeleteMemberDialog = ({ id, open, setOpen }: Props): JSX.Element => {
   const { t: translateAccount } = useAccountTranslation();
-  const [open, setOpen] = useState(false);
+  const [confirmationDeleteValue, setConfirmationDeleteValue] = useState('');
+
   const { mutate: deleteMember } = mutations.useDeleteMember();
 
   const alertDialogTitle = 'alert-dialog-title';
   const alertDialogDescription = 'alert-dialog-description';
```
Comment (spaenleh) [inline src/components/main/DeleteMemberDialog.tsx:29]: This should be translated and maybe a bit more descriptive

File: src/components/main/DeleteAccount.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+  if (member) {
+    return (
+      <>
+        <Grid my={1}>
+          <Typography variant="h5">
+            {t('PROFILE_DELETE_ACCOUNT_TITLE')}
+          </Typography>
+          <StyledGrid item>
+            <Typography variant="body2" mt={1}>
```
Comment (spaenleh) [inline src/components/main/DeleteAccount.tsx:34]: Remove `mt` here.

**Review by spaenleh — CHANGES_REQUESTED** [review]
Thank you for these improvements. 💪 I tried different things locally. I have pushed my changes so you can play with them and see them in context as well as see the exact diff on the `86-delete-account-proposal` branch.

Overall I think we are almost there, I mostly have issues and comments on the style of code. I think it can be more concise and have less specific style overrides (I am sorry but I am not a fan of margin overrides).

So please have a look and let me know if you have comments, questions or if you think my opinions should be challenged (they should).

**Review by spaenleh — APPROVED** [review]
Nice ! There is just the tiny issue of the break line inside the translation which should be fixed before merging. 💪

**Review by spaenleh — APPROVED** [review]

**Pull request merged**